### ISSUE TITLE
Error accessing identity name with windows authentication

### ISSUE THREAD

**Issue by FTWinston** [aspnet/SignalR#2948]
* Using Microsoft.AspNetCore.SignalR 1.0.1 on .NET Core 2.1, with @aspnet/signalr 1.0.3.
* Javascript client, running in Chrome v68 & v69 on Windows 10.
* Hosting either in IIS 8 on Windows Server 2012 R2, or in IIS express on Windows 10.

Using windows authentication and long polling, I'm getting an ObjectDisposedException when accessing Context.User.Identity.Name from inside any hub methods, or from OnConnectedAsync.

This has only started happening since upgrading from .NET Core 2.0 and SignalR 1.0.0-alpha2-final.

It doesn't happen every time, but if you have connected succesfully once before, it always happens on your second connection attempt.

```
System.ObjectDisposedException: Safe handle has been closed
   at System.Runtime.InteropServices.SafeHandle.DangerousAddRef(Boolean& success)
   at System.StubHelpers.StubHelpers.SafeHandleAddRef(SafeHandle pHandle, Boolean& success)
   at Interop.Advapi32.GetTokenInformation(SafeAccessTokenHandle TokenHandle, UInt32 TokenInformationClass, SafeLocalAllocHandle TokenInformation, UInt32 TokenInformationLength, UInt32& ReturnLength)
   at System.Security.Principal.WindowsIdentity.GetTokenInformation(SafeAccessTokenHandle tokenHandle, TokenInformationClass tokenInformationClass, Boolean nullOnInvalidParam)
   at System.Security.Principal.WindowsIdentity.get_User()
   at System.Security.Principal.WindowsIdentity.<GetName>b__46_0()
   at System.Security.Principal.WindowsIdentity.<>c__DisplayClass62_0.<RunImpersonatedInternal>b__0(Object <p0>)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location where exception was thrown ---
   at System.Security.Principal.WindowsIdentity.RunImpersonatedInternal(SafeAccessTokenHandle token, Action action)
   at System.Security.Principal.WindowsIdentity.RunImpersonated(SafeAccessTokenHandle safeAccessTokenHandle, Action action)
   at System.Security.Principal.WindowsIdentity.GetName()
   at System.Security.Principal.WindowsIdentity.get_Name()
   at Site.Hubs.Changes.AnyMethod(Int32 changeID) in C:\Site\Hubs\Changes.cs:line 94
```

Here are my diagnostic logs: [server log](https://github.com/aspnet/SignalR/files/2375614/server.log.txt) & [client log](https://github.com/aspnet/SignalR/files/2375613/client.log.txt)

This exception doesn't occur when using SSE instead of long polling, though I haven't yet got that to work with IIS. It looks like there is [finally some movement](https://bugs.chromium.org/p/chromium/issues/detail?id=423609) on chrome support for windows auth 
on websockets, so I may be able to switch to those eventually.

When this does occur, I've been able to work around it by doing the following, but this is hardly a sensible fix:
```
while (true)
{
    try
    {
        return Context.User.Identity.Name;
    }
    catch (ObjectDisposedException)
    {
        await Task.Delay(100);
    }
}
```

Alternatively, it works pretty consistently if I put `await Task.Delay(1000);` at the start of my hub method.

Any better suggestions? It looks to be related to [this bug](https://github.com/aspnet/Logging/issues/543), at least superficially.

**Comment by anurse** [comment]
I think I know the source of your issue. Because Long Polling runs over several requests, the Identity you get ends up getting disposed while your Hub is still processing. This won't occur with SSE or WebSockets because there is a single long-running request.

This is going to be a tricky one to resolve I'm afraid. We'll put it in our queue and see what we can do. I'll also update the docs since with this issue, I don't think we can recommend Windows Auth for Long Polling requests at all.

@davidfowl This is the cloned HttpContext wreaking some havoc.

> This exception doesn't occur when using SSE instead of long polling, though I haven't yet got that to work with IIS

Could you file a new bug (don't want to confuse this issue) for this and elaborate further? We definitely test SSE with IIS so this should be something that works and it's especially important if Long Polling doesn't support Windows Auth!

**Comment by anurse** [comment]
Ok, after talking with @Tratcher (resident Windows Auth expert) this might be a lot easier than I'd feared to fix. We just need to make sure to clone the WindowsIdentity properly.

**Comment by FTWinston** [comment]
Thanks @anurse, I've raised ticket #2954 for my SSE problem, although I suspect it's likely to be a configuration issue.

Let me know if I can be of help with reproducing this error.

**Comment by BrennanConroy** [comment]
This will be fixed in the 1.1.0-preview3 release
https://github.com/aspnet/SignalR/commit/f0a34a4ca46129b8434020dedb8c360e751e2963
